src/test/scala/org/ergoplatform/network/DecodingUtils.scala

```diff
@@ -0,0 +1,61 @@
+package org.ergoplatform.network
+
+import java.nio.ByteBuffer
+
+// Common functions used in parsers, collected in one place to be easily understood and rewritten into other languages
+trait DecodingUtils {
+
+  /**
+    * Read unsigned byte
+    */
+  def getUByte(buf: ByteBuffer): Int = getByte(buf) & 0xFF
+
+  /**
+    * Read signed byte
+    */
+  def getByte(buf: ByteBuffer): Byte = buf.get
+
+  /**
+    * Reading up to 64 bits VLQ-encoded value. The value is considered to be signed.
+    * Can be read less than 64 bytes!
+    * See https://en.wikipedia.org/wiki/Variable-length_quantity
+    */
+  def getULong(buf: ByteBuffer): Long = {
+    var result: Long = 0
+    var shift = 0
+    while (shift < 64) {
+      val b = getByte(buf)
+      result = result | ((b & 0x7F).toLong << shift)
+      if ((b & 0x80) == 0) return result
+      shift += 7
+    }
+    throw new IllegalStateException("Trying to read long, but more bytes than needed found")
+  }
```

Comment:
This duplicates code already present in scorex-util, which is in Ergo's dependencies. https://github.com/ScorexFoundation/scorex-util/blob/master/src/main/scala/scorex/util/serialization/VLQReader.scala